### PR TITLE
z_lib fully matched

### PR DESCRIPTION
**Pull request by Rozelette**
Also updated the documentation to OOT and named the new MM functions

### CODE REVIEW

File: include/stdint.h [hunk 1]
```diff
@@ -0,0 +1,29 @@
+#ifndef _STDINT_H_
+#define _STDINT_H_
+
+#include <ultra64.h>
+
+typedef long intptr_t;
+typedef unsigned long uintptr_t;
```
Comment (Thar0) [inline include/stdint.h:7]:
```suggestion
typedef s32 intptr_t;
typedef u32 uintptr_t;
```
Unfortunately IDO cares about `long` vs `int` for matching, so to be completely safe this should mirror the 32-bit types in `ultratypes.h`, at least for now.

**Review by Thar0 — COMMENTED** [review]
The file itself looks good to me. Just some thoughts.

File: include/functions.h [hunk 2]
```diff
@@ -1836,40 +1836,40 @@ void Math_Vec3f_Scale(Vec3f* vec, f32 scale);
...
+void func_801000A4(u16 sfxId);
+void func_801000CC(u16 sfxId);
+void func_801000F4(s32 a0, u16 a1);
+void Lib_Vec3f_TranslateAndRotateY(Vec3f* translation, s16 a, Vec3f* src, Vec3f* dst);
 void Lib_LerpRGB(Color_RGB8* a, Color_RGB8* b, f32 t, Color_RGB8* dst);
-f32 Lib_PushAwayVec3f(Vec3f* start, Vec3f* pusher, f32 distanceToApproach);
+f32 Math_Vec3f_StepTo(Vec3f* start, Vec3f* target, f32 speed);
 void Lib_Nop801004FC(void);
 void* Lib_PtrSegToVirt(void* ptr);
 void* Lib_PtrSegToVirtNull(void* ptr);
```
Comment (Thar0) [inline include/functions.h:1875]: Additionally, this is solely my opinion but I believe a better naming scheme for these functions (as well as the `PtrSegToK0` functions just below these) would be names that directly mirror the macros. i.e. `Lib_SegmentedToVirtual` and `Lib_PhysicalToVirtual`

File: include/stdint.h (hunk 1 above)
Comment (fig02) [reply include/stdint.h:7]: any reason this is needed on top of ultratypes?

Comment (m4xw) [reply include/stdint.h:7]: grafik

**Comment by Kenix3** [comment]
Looks good to me!

**Pull request merged**